### ISSUE TITLE
[e2e conformance] k8s.gcr.io/conformance-arm64:v1.19.4 (f2676fc8aea6): The Architecture of v1.19.4 shows amd64

### ISSUE THREAD

**Issue by tanjing2020** [kubernetes/kubernetes#102315]
#### What happened:
```
root@node4:/usr1# docker images | grep v1.19.4
k8s.gcr.io/conformance-arm64         v1.19.4             f2676fc8aea6        6 months ago        217MB
k8s.gcr.io/conformance               v1.19.4             1bea13400cee        6 months ago        230MB
root@node4:/usr1# docker inspect f2676fc8aea6 | grep Architecture
        "Architecture": "amd64",
root@node4:/usr1# docker inspect 1bea13400cee | grep Architecture
        "Architecture": "amd64",
```

#### What you expected to happen:
k8s.gcr.io/conformance-arm64:v1.19.4 is the arm architecture mirror

#### How to reproduce it (as minimally and precisely as possible):
docker pull k8s.gcr.io/conformance-arm64:v1.19.4

#### Anything else we need to know?:
k8s.gcr.io/conformance:v1.19.0 also has the same problem

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```
NAME="EulerOS"
VERSION="2.0 (SP8)"
ID="euleros"
ID_LIKE="rhel fedora centos"
VERSION_ID="2.0"
PRETTY_NAME="EulerOS 2.0 (SP8)"
ANSI_COLOR="0;31"
```
- Kernel (e.g. `uname -a`):
```
Linux kwephicprd61197 4.19.36-vhulk1907.1.0.h453.eulerosv2r8.aarch64 #1 SMP Mon Jul 22 00:00:00 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by pacoxu** [comment]
The way that I used to check image architecture is like 
```
curl -s https://k8s.gcr.io/v2/conformance/manifests/v1.19.4
```

**Comment by tanjing2020** [comment]
```
root@node4:/usr1# curl -s https://k8s.gcr.io/v2/conformance/manifests/v1.19.4
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1792,
         "digest": "sha256:24ee4a661e0dfcf653f1de4889472a2784f3b74ac67c2c19f6dbd9bf8ab92c0f",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1792,
         "digest": "sha256:80854b29ab548d0680728979e88aa78b0567007f4cf2cfef73707c1765e97b34",
         "platform": {
            "architecture": "arm",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1792,
         "digest": "sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1792,
         "digest": "sha256:5b153b79486910487e9121cdcc90ab79c9952d91c445c41027802768205c8ae0",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1792,
         "digest": "sha256:00827863ae3271818c48dca53cc5b2c48e50d8ec58a12fb15f6ec0b3331a5e20",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}root@node4:/usr1#
```
```
root@node3:~# docker images
REPOSITORY                                   TAG                 IMAGE ID            CREATED             SIZE
hello-world                                  latest              d1165f221234        2 months ago        13.3kB
k8s.gcr.io/pause                             3.4.1               0f8457a4c2ec        4 months ago        683kB
k8s.gcr.io/coredns/coredns                   v1.8.0              296a6d5035e2        7 months ago        42.5MB
nfvpe/sriov-device-plugin                    v3.1                575969f9dfbe        18 months ago       25.3MB
gcr.io/kubernetes-e2e-test-images/dnsutils   1.3                 96572c6d9c9c        23 months ago       64.6MB
nginx                                        1.14-alpine         8a2fb25a19f5        2 years ago         16MB
tutum/dnsutils                               latest              6cd78a6d3256        6 years ago         200MB
nginx                                        1.7.9               84581e99d807        6 years ago         91.7MB
root@node3:~# docker pull k8s.gcr.io/conformance@sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767
sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767: Pulling from conformance
babf97a3f00a: Pull complete
4508a7979cb3: Pull complete
4876a5599c90: Pull complete
dfdae211448d: Pull complete
77b87db2a9b9: Pull complete
960d92c2c8f2: Pull complete
1f96739384aa: Pull complete
Digest: sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767
Status: Downloaded newer image for k8s.gcr.io/conformance@sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767
k8s.gcr.io/conformance@sha256:28742e3fa437c99b20a125491562a7481b0a9bca8d8af12b05c2669ad8cee767
root@node3:~# docker images
REPOSITORY                                   TAG                 IMAGE ID            CREATED             SIZE
hello-world                                  latest              d1165f221234        2 months ago        13.3kB
k8s.gcr.io/pause                             3.4.1               0f8457a4c2ec        4 months ago        683kB
k8s.gcr.io/conformance                       <none>              f2676fc8aea6        6 months ago        217MB
k8s.gcr.io/coredns/coredns                   v1.8.0              296a6d5035e2        7 months ago        42.5MB
nfvpe/sriov-device-plugin                    v3.1                575969f9dfbe        18 months ago       25.3MB
gcr.io/kubernetes-e2e-test-images/dnsutils   1.3                 96572c6d9c9c        23 months ago       64.6MB
nginx                                        1.14-alpine         8a2fb25a19f5        2 years ago         16MB
tutum/dnsutils                               latest              6cd78a6d3256        6 years ago         200MB
nginx                                        1.7.9               84581e99d807        6 years ago         91.7MB

```
```
root@node3:~# docker inspect f2676fc8aea6 | grep Architecture
        "Architecture": "amd64",

```

**Comment by tanjing2020** [comment]
```
[root@kwephicprd61197 paas]# docker run -it f2676fc8aea6 sh
standard_init_linux.go:220: exec user process caused "exec format error"
libcontainer: container start initialization failed: standard_init_linux.go:220: exec user process caused "exec format error"
```
```
[root@kwephicprd61197 paas]# file /lib/systemd/systemd
/lib/systemd/systemd: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, BuildID[sha1]=1d0febb1c36f1950b8d07a4fad3371d4fc4588b2, stripped
```

**Comment by vielmetti** [comment]
I'm able to confirm this problem on an arm64 Ubuntu 20.04 system.

```
ed@e-jet:~$ sudo docker run -it f2676fc8aea6 sh
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:219: exec user process caused: exec format error
```

**Comment by vielmetti** [comment]
Preiously: https://github.com/kubernetes/kubernetes/issues/98086

**Comment by BenTheElder** [comment]
What about 1.19.11? We are not going to change the contents of an old image (see #98086) and 1.19.4 is several patch releases back now. If any fixes landed they might only be in a more recent image. I've lost track of what is landed in 1.19 (we're developing 1.22...)

**Comment by vielmetti** [comment]
1.19.11 gives me this:

```
ed@e-jet:~$ sudo docker run -it k8s.gcr.io/conformance-arm64:v1.19.11 /bin/sh
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:219: exec user process caused: exec format error
```

**Comment by vielmetti** [comment]
1.21.1 gives me a nominally successful arm64 image! (at least I can launch it without error)

**Comment by BenTheElder** [comment]
https://console.cloud.google.com/gcr/images/k8s-artifacts-prod/US/conformance-arm@sha256:7cc6a33e0380a831bd1e4d04ba60a638fd07cdacd85a2142a0515e00c117aaac/details?tab=info is a manifest list ... 🧐

But the multi arch image looks right at a glance?
https://console.cloud.google.com/gcr/images/k8s-artifacts-prod/US/conformance@sha256:f45588081cc6d91fa66a0c36a3bc23004a81f84b55787a834eb911d3159d77a4/details?tab=info

**Comment by vielmetti** [comment]
At a glance, the multiarch 1.19.11 version fails to load on arm64, but the multiarch 1.21.1 is fine.

Details:

```
ed@e-jet:~$ sudo docker run -it us.gcr.io/k8s-artifacts-prod/conformance:v1.19.11 /bin/sh
Unable to find image 'us.gcr.io/k8s-artifacts-prod/conformance:v1.19.11' locally
v1.19.11: Pulling from k8s-artifacts-prod/conformance
Digest: sha256:f45588081cc6d91fa66a0c36a3bc23004a81f84b55787a834eb911d3159d77a4
Status: Downloaded newer image for us.gcr.io/k8s-artifacts-prod/conformance:v1.19.11
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:219: exec user process caused: exec format error
ed@e-jet:~$ sudo docker run -it us.gcr.io/k8s-artifacts-prod/conformance:v1.21.1 /bin/sh
Unable to find image 'us.gcr.io/k8s-artifacts-prod/conformance:v1.21.1' locally
v1.21.1: Pulling from k8s-artifacts-prod/conformance
Digest: sha256:c80a52450de5efb23e93d9c7636f7959177b2a0b3cd14f0fca9b2ab898b42511
Status: Downloaded newer image for us.gcr.io/k8s-artifacts-prod/conformance:v1.21.1
# ls
e2e.test  ginkgo  kubectl
# ls -ltr
total 198532
-rwxr-xr-x 1 root root  44105728 May 12 14:29 kubectl
-rwxr-xr-x 1 root root   6881280 May 12 14:29 ginkgo
-rwxr-xr-x 1 root root 152301876 May 12 14:29 e2e.test
# pwd
/usr/local/bin
#
```

**Comment by BenTheElder** [comment]
I think that's probably it for this bug then, I don't think backporting a change that changes build tools will be approved (you're welcome to try and see if I'm wrong though! I'm not actually sure!) and this is fixed in subsequent releases.

we also will not change an existing image, but if it was backported a future 1.19.X could contain this change.

**Comment by saschagrunert** [comment]
Hey @vielmetti, thank you for the report! We explicitly decided  to not backport the fixes to the conformance image to release-1.19 to not silently break downstream consumers: https://github.com/kubernetes/kubernetes/pull/102126

Closing this for now, feel free to reopen if there is anything missing.

/close
